Revit_oM/Misc/RevitGeometry.cs

```diff
@@ -20,50 +20,43 @@
  * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
  */
 
-using System.ComponentModel;
-
 using BH.oM.Base;
-using BH.oM.Reflection.Attributes;
+using BH.oM.Geometry;
+using System.Collections.Generic;
+using System.ComponentModel;
 
-namespace BH.Engine.Adapters.Revit
+namespace BH.oM.Adapters.Revit
 {
-    public static partial class Query
+    [Description("Fragment containing the geometry extracted from Revit element represented by the BHoM object.")]
+    public class RevitGeometry : IFragment, IImmutable
     {
         /***************************************************/
-        /****              Public methods               ****/
+        /****             Public Properties             ****/
         /***************************************************/
 
-        [Deprecated("3.2", "BH.Engine.Adapters.Revit.Query.SpaceId is not used any more.")]
-        [Description("Gets integer representation of ElementId of Revit space element correspondent to given BHoMObject. This value is stored in CustomData under key SpaceID.")]
-        [Input("bHoMObject", "BHoMObject to be queried.")]
-        [Output("elementId")]
-        public static int SpaceId(this IBHoMObject bHoMObject)
-        {
-            if (bHoMObject == null)
-                return -1;
+        [Description("Edge curves of Revit element represented by the BHoM object carrying this fragment.")]
+        public virtual List<ICurve> Edges { get; set; } = null;
+
+        [Description("Surface geometry of Revit element represented by the BHoM object carrying this fragment.")]
+        public virtual List<ISurface> Surfaces { get; set; } = null;
+
+        [Description("Meshed surfaces of Revit element represented by the BHoM object carrying this fragment.")]
+        public virtual List<Mesh> Meshes { get; set; } = null;
 
-            object value = null;
-            if (bHoMObject.CustomData.TryGetValue("SpaceID", out value))
-            {
-                if (value is string)
-                {
-                    int num = -1;
-                    if (int.TryParse((string)value, out num))
-                        return num;
-                }
-                else if (value is int)
-                {
-                    return (int)value;
-                }
-                else
-                {
-                    return -1;
-                }
-            }
 
-            return -1;
+        /***************************************************/
+        /****            Public Constructors            ****/
+        /***************************************************/
+
+        public RevitGeometry(List<ICurve> edges, List<ISurface> surfaces, List<Mesh> meshes)
+        {
+            Edges = edges;
+            Surfaces = surfaces;
+            Meshes = meshes;
```

Comment:
If a switch is made to ReadOnlyCollection:

```suggestion
            Edges = edges != null ? new ReadOnlyCollection<ICurve>(edges) : null;
            Surfaces = surfaces != null ? new ReadOnlyCollection<ISurface>(surfaces) : null;
            Meshes = meshes != null ? new ReadOnlyCollection<Mesh>(meshes) : null;
```